From 298557ddf7068a4c86266131afc1201712227f35 Mon Sep 17 00:00:00 2001
From: Edmunt Pienkowsky <roed@onet.eu>
Date: Mon, 6 Apr 2020 12:06:43 +0200
Subject: [PATCH 2/3] Frontend - build without source map


diff --git a/scripts/webpack/webpack.prod.js b/scripts/webpack/webpack.prod.js
index f4c6dc4f40..f6b8c3f0d2 100644
--- a/scripts/webpack/webpack.prod.js
+++ b/scripts/webpack/webpack.prod.js
@@ -11,7 +11,6 @@ const OptimizeCSSAssetsPlugin = require('optimize-css-assets-webpack-plugin');
 
 module.exports = merge(common, {
   mode: 'production',
-  devtool: 'source-map',
 
   entry: {
     dark: './public/sass/grafana.dark.scss',
@@ -79,8 +78,7 @@ module.exports = merge(common, {
     minimizer: [
       new TerserPlugin({
         cache: false,
-        parallel: false,
-        sourceMap: true,
+        parallel: false
       }),
       new OptimizeCSSAssetsPlugin({}),
     ],
