From 150e2e6365cc41741082b24975751c79cb2d38a0 Mon Sep 17 00:00:00 2001
From: Edmunt Pienkowsky <roed@onet.eu>
Date: Fri, 8 Oct 2021 11:06:08 +0200
Subject: [PATCH] build: setup improvements


diff --git a/pkg/build/cmd.go b/pkg/build/cmd.go
index 5817cee065..4f892b533f 100644
--- a/pkg/build/cmd.go
+++ b/pkg/build/cmd.go
@@ -64,7 +64,9 @@ func RunCmd() int {
 	for _, cmd := range flag.Args() {
 		switch cmd {
 		case "setup":
-			setup(opts.goos)
+			for _, binary := range binaries {
+				doSetup(opts.goos, "./pkg/cmd/"+binary)
+			}
 
 		case "build-srv", "build-server":
 			if !opts.isDev {
@@ -158,12 +160,12 @@ func genPackageVersion(version string, iteration string) string {
 	}
 }
 
-func setup(goos string) {
+func doSetup(goos string, binaryName string) {
 	args := []string{"install", "-v"}
 	if goos == GoOSWindows {
 		args = append(args, "-buildmode=exe")
 	}
-	args = append(args, "./pkg/cmd/grafana-server")
+	args = append(args, binaryName)
 	runPrint("go", args...)
 }
 
diff --git a/pkg/build/git.go b/pkg/build/git.go
index 6c86d74e00..e294edd8b0 100644
--- a/pkg/build/git.go
+++ b/pkg/build/git.go
@@ -1,5 +1,9 @@
 package build
 
+import (
+	"os"
+)
+
 func getGitBranch() string {
 	v, err := runError("git", "rev-parse", "--abbrev-ref", "HEAD")
 	if err != nil {
@@ -11,7 +15,7 @@ func getGitBranch() string {
 func getGitSha() string {
 	v, err := runError("git", "rev-parse", "--short", "HEAD")
 	if err != nil {
-		return "unknown-dev"
+		return os.Getenv("GRAFANA_COMMIT")
 	}
 	return string(v)
 }
