From d52144f38e2948ef045d97b1c6f3a2089be8fb97 Mon Sep 17 00:00:00 2001
From: Edmunt Pienkowsky <roed@onet.eu>
Date: Sat, 15 Feb 2020 08:49:26 +0100
Subject: [PATCH 1/2] build: setup improvements


diff --git a/build.go b/build.go
index 4fcb08e6d8..0fdcde0b67 100644
--- a/build.go
+++ b/build.go
@@ -108,7 +108,9 @@ func main() {
 	for _, cmd := range flag.Args() {
 		switch cmd {
 		case "setup":
-			setup()
+			for _, binary := range binaries {
+				doSetup("./pkg/cmd/"+binary)
+			}
 
 		case "build-srv", "build-server":
 			clean()
@@ -218,12 +220,12 @@ func genPackageVersion() string {
 	}
 }
 
-func setup() {
+func doSetup(binaryName string) {
 	args := []string{"install", "-v"}
 	if goos == windows {
 		args = append(args, "-buildmode=exe")
 	}
-	args = append(args, "./pkg/cmd/grafana-server")
+	args = append(args, binaryName)
 	runPrint("go", args...)
 }
 
@@ -364,7 +366,7 @@ func getGitBranch() string {
 func getGitSha() string {
 	v, err := runError("git", "rev-parse", "--short", "HEAD")
 	if err != nil {
-		return "unknown-dev"
+		return os.Getenv("GRAFANA_COMMIT")
 	}
 	return string(v)
 }
