From e5045cd76e0475e87e3dd59f8ab2cd914ce4e29b Mon Sep 17 00:00:00 2001
From: Edmunt Pienkowsky <roed@onet.eu>
Date: Sat, 15 Feb 2020 08:49:26 +0100
Subject: [PATCH 1/4] build: allow to specify commit hash

via GRAFANA_COMMIT environment variable.

diff --git a/build.go b/build.go
index 525f8a55a7..30295c3537 100644
--- a/build.go
+++ b/build.go
@@ -595,7 +595,7 @@ func getGitBranch() string {
 func getGitSha() string {
 	v, err := runError("git", "rev-parse", "--short", "HEAD")
 	if err != nil {
-		return "unknown-dev"
+		return os.Getenv("GRAFANA_COMMIT")
 	}
 	return string(v)
 }
-- 
2.27.0

