From a25fc70c79c58c35313d2b240b33b73cdc52dd48 Mon Sep 17 00:00:00 2001
From: Edmunt Pienkowsky <roed@onet.eu>
Date: Sat, 15 Feb 2020 08:49:26 +0100
Subject: [PATCH 2/4] build: allow to specify commit hash

via GRAFANA_COMMIT environment variable.

diff --git a/build.go b/build.go
index b947be43a0..9c57873771 100644
--- a/build.go
+++ b/build.go
@@ -606,7 +606,7 @@ func getGitBranch() string {
 func getGitSha() string {
 	v, err := runError("git", "rev-parse", "--short", "HEAD")
 	if err != nil {
-		return "unknown-dev"
+		return os.Getenv("GRAFANA_COMMIT")
 	}
 	return string(v)
 }
-- 
2.26.2

