From 3fa5a202ff79f98b40392acc753c9b5d064d618f Mon Sep 17 00:00:00 2001
From: Edmunt Pienkowsky <roed@onet.eu>
Date: Thu, 4 Jun 2020 09:54:05 +0200
Subject: [PATCH 2/4] Reproducible build


diff --git a/build.go b/build.go
index 30295c3537..199fda877f 100644
--- a/build.go
+++ b/build.go
@@ -502,6 +502,7 @@ func build(binaryName, pkg string, tags []string) {
 		args = append(args, "-race")
 	}
 
+	args = append(args, "-trimpath")
 	args = append(args, "-o", binary)
 	args = append(args, pkg)
 
@@ -537,6 +538,7 @@ func ldflags() string {
 	if v := os.Getenv("LDFLAGS"); v != "" {
 		b.WriteString(fmt.Sprintf(" -extldflags \"%s\"", v))
 	}
+	b.WriteString(" -buildid=")
 	return b.String()
 }
 
-- 
2.27.0

