From fe8769c792ab17fe032b83a17dcd04df3dc1d8e7 Mon Sep 17 00:00:00 2001
From: Edmunt Pienkowsky <roed@onet.eu>
Date: Thu, 4 Jun 2020 09:54:05 +0200
Subject: [PATCH 2/3] Reproducible build


diff --git a/build.go b/build.go
index 26481038c5..8c725e9b85 100644
--- a/build.go
+++ b/build.go
@@ -489,6 +489,7 @@ func doBuild(binaryName, pkg string, tags []string) {
 		args = append(args, "-race")
 	}
 
+	args = append(args, "-trimpath")
 	args = append(args, "-o", binary)
 	args = append(args, pkg)
 
@@ -524,6 +525,7 @@ func ldflags() string {
 	if v := os.Getenv("LDFLAGS"); v != "" {
 		b.WriteString(fmt.Sprintf(" -extldflags \"%s\"", v))
 	}
+	b.WriteString(" -buildid=")
 	return b.String()
 }
 
-- 
2.28.0

