From e9ebe1626e8c0cd8f022750302c0a74809e6eaf4 Mon Sep 17 00:00:00 2001
From: Edmunt Pienkowsky <roed@onet.eu>
Date: Fri, 29 Mar 2019 10:51:10 +0100
Subject: [PATCH 2/3] build: allow to specify commit hash

Via environment variable GRAFANA_COMMIT.

Signed-off-by: Edmunt Pienkowsky <roed@onet.eu>

diff --git a/build.go b/build.go
index e85f7fe0d..9f1f81caa 100644
--- a/build.go
+++ b/build.go
@@ -589,7 +589,7 @@ func getGitBranch() string {
 func getGitSha() string {
 	v, err := runError("git", "rev-parse", "--short", "HEAD")
 	if err != nil {
-		return "unknown-dev"
+		return os.Getenv("GRAFANA_COMMIT")
 	}
 	return string(v)
 }
-- 
2.23.0

