From 6261c3d267743e9693038516d46443fbcb5ec46c Mon Sep 17 00:00:00 2001
From: Edmunt Pienkowsky <roed@onet.eu>
Date: Fri, 29 Mar 2019 10:51:10 +0100
Subject: [PATCH 2/2] build: allow to specify commit hash

Via environment variable GRAFANA_COMMIT.

Signed-off-by: Edmunt Pienkowsky <roed@onet.eu>

diff --git a/build.go b/build.go
index b947be43a..9c5787377 100644
--- a/build.go
+++ b/build.go
@@ -606,7 +606,7 @@ func getGitBranch() string {
 func getGitSha() string {
 	v, err := runError("git", "rev-parse", "--short", "HEAD")
 	if err != nil {
-		return "unknown-dev"
+		return os.Getenv("GRAFANA_COMMIT")
 	}
 	return string(v)
 }
-- 
2.24.1

