From c2b91832c60c0a554fbcea866b35d50de9cde1ae Mon Sep 17 00:00:00 2001
From: Edmunt Pienkowsky <roed@onet.eu>
Date: Fri, 29 Mar 2019 10:51:10 +0100
Subject: [PATCH 2/3] build: allow to specify commit hash

Via environment variable GRAFANA_COMMIT.

Signed-off-by: Edmunt Pienkowsky <roed@onet.eu>

diff --git a/build.go b/build.go
index 6cfa79ea7..bcc80c828 100644
--- a/build.go
+++ b/build.go
@@ -586,7 +586,7 @@ func getGitBranch() string {
 func getGitSha() string {
 	v, err := runError("git", "rev-parse", "--short", "HEAD")
 	if err != nil {
-		return "unknown-dev"
+		return os.Getenv("GRAFANA_COMMIT")
 	}
 	return string(v)
 }
-- 
2.23.0

